crypto/src/dex/swap/payload.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::anyhow;
+use penumbra_proto::core::dex::v1alpha1 as pb;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{ka, FullViewingKey};
+
+use super::{SwapCiphertext, SwapPlaintext};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(try_from = "pb::SwapPayload", into = "pb::SwapPayload")]
+pub struct SwapPayload {
+    pub commitment: penumbra_tct::Commitment,
+    pub ephemeral_key: ka::Public,
+    pub encrypted_swap: SwapCiphertext,
+}
```

Comment:
This is the analogue of the `EncryptedNote`. It's named this way because of the realization that, after all that refactoring and renaming, it might actually be best to call the `EncryptedNote` a `NotePayload` after all, so that we have "state payloads" which are an enum of "rolled up data", "note payloads", "swap payloads", etc.